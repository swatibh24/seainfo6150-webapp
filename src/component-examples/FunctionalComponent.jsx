import React from "react";
import PropTypes from "prop-types";

const FunctionalComponent = ({ name = "World" }) => {
  return <div>Hello {name}, here&rsquo;s my functional component</div>;
};

FunctionalComponent.propTypes = {
  name: PropTypes.string.isRequired
};
export default FunctionalComponent;
