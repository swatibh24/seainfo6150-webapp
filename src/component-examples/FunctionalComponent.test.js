import React from "react";
import FunctionalComponent from "./FunctionalComponent";

describe("FunctionalComponent tests", () => {
  it("renders correctly", () => {
    const { container } = render(<FunctionalComponent />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly when passed a name prop", () => {
    const { container } = render(<FunctionalComponent name="April" />);
    expect(container).toMatchSnapshot();
  });
});
