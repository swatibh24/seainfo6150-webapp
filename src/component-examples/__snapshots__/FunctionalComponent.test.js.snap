// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalComponent tests renders correctly 1`] = `
<div>
  <div>
    Hello 
    World
    , here’s my functional component
  </div>
</div>
`;

exports[`FunctionalComponent tests renders correctly when passed a name prop 1`] = `
<div>
  <div>
    Hello 
    April
    , here’s my functional component
  </div>
</div>
`;
